src/FluxCalculator.jl

```diff
@@ -91,4 +100,296 @@ function calculate_surface_air_density(atmos_sim::Interfacer.AtmosModelSimulatio
+                    F_turb_ρτxz, F_turb_ρτyz, F_shf, F_lhf, F_turb_moisture =
+                        get_surface_fluxes_point!(inputs, surface_params)
+
+                    fields = (;
+                        F_turb_ρτxz = F_turb_ρτxz,
+                        F_turb_ρτyz = F_turb_ρτyz,
+                        F_turb_energy = F_shf .+ F_lhf,
+                        F_turb_moisture = F_turb_moisture,
+                    )
+
+                    # update the fluxes of this surface model
+                    update_turbulent_fluxes_point!(sim, fields, colidx)
+
+                    # add the flux contributing from this surface
+                    @. csf.F_turb_ρτxz[colidx] += F_turb_ρτxz * area_fraction * area_mask
+                    @. csf.F_turb_ρτyz[colidx] += F_turb_ρτyz * area_fraction * area_mask
+                    @. csf.F_turb_energy[colidx] += (F_shf .+ F_lhf) * area_fraction * area_mask
+                    @. csf.F_turb_moisture[colidx] += F_turb_moisture * area_fraction * area_mask
+
+                end
+            end
+        end
+
+    end
+
+    # TODO: add allowable bounds here, check explicitly that all fluxes are equal
+    # @assert(extrema(csf.F_turb_energy) ≈ (-1000.0, 1000.0))
+
+end
+
+abstract type AbstractSurfaceFluxScheme end
+struct BulkScheme <: AbstractSurfaceFluxScheme end
+struct MoninObukhovScheme <: AbstractSurfaceFluxScheme end
+
+"""
+    get_scheme_specific_properties(scheme::AbstractSurfaceFluxScheme, sim::Interfacer.SurfaceModelSimulation, colidx::Fields.ColumnIndex)
+
+Returns the scheme-specific properties for the surface model simulation `sim`.
+"""
+function get_scheme_specific_properties(
+    ::BulkScheme,
+    sim::Interfacer.SurfaceModelSimulation,
+    colidx::Fields.ColumnIndex,
+)
+    Ch = Interfacer.get_field(sim, Val(:heat_transfer_coefficient), colidx)
+    Cd = Interfacer.get_field(sim, Val(:beta), colidx)
+    beta = Interfacer.get_field(sim, Val(:drag_coefficient), colidx)
+    FT = eltype(Ch)
+    return (; z0b = FT(0), z0m = FT(0), Ch = Ch, Cd = Cd, beta = beta, gustiness = FT(1))
+end
+function get_scheme_specific_properties(
+    ::MoninObukhovScheme,
+    sim::Interfacer.SurfaceModelSimulation,
+    colidx::Fields.ColumnIndex,
+)
+    z0m = Interfacer.get_field(sim, Val(:roughness_momentum), colidx)
+    z0b = Interfacer.get_field(sim, Val(:roughness_buoyancy), colidx)
+    beta = Interfacer.get_field(sim, Val(:beta), colidx)
+    FT = eltype(z0m)
+    return (; z0b = z0b, z0m = z0m, Ch = FT(0), Cd = FT(0), beta = beta, gustiness = FT(1))
+end
+
+"""
+    surface_inputs(scheme::AbstractSurfaceFluxScheme, thermo_state_sfc, thermo_state_int, uₕ_int, z_int, z_sfc, z0b, z0m, Ch, Cd, beta, gustiness)
+
+Returns the inputs for the surface model simulation `sim`.
+"""
+function surface_inputs(
+    ::BulkScheme,
+    thermo_state_sfc,
+    thermo_state_int,
+    uₕ_int,
+    z_int,
+    z_sfc,
+    z0b,
+    z0m,
+    Ch,
+    Cd,
+    beta,
+    gustiness,
```

Comment:
Could we package these as a NamedTuple and pass that around for readability? I think it would reduce the line count for methods of this function a lot

Reply:
Good idea! I'm now unpacking the surface properties inside the `surface_inputs` function.